src/aw-client.ts

```diff
@@ -0,0 +1,103 @@
+import axios, { AxiosInstance } from 'axios';
+
+// declare var window: any;
+declare var module: any;
+
+// let isBrowser: boolean = (window !== undefined);
+const isNode = (typeof module !== 'undefined' && module.exports);
+     
+
+class AWClient {
+    public clientname: string;
+    public testing: boolean;
+    public req: AxiosInstance;
+
+    constructor(clientname: string, testing: boolean, baseurl: string) {
+        this.clientname = clientname;
+        this.testing = testing;
+        if (baseurl == undefined){
+            let port = !testing ? 5600 : 5666;
+            baseurl = 'http://127.0.0.1:'+port;
+        }
+
+        this.req = axios.create({
+          baseURL: baseurl+'/api',
+          timeout: 10000,
+          headers: (!isNode) ? {} : {'User-Agent': 'aw-client-js/0.1'}
+        });
+
+        // Make 304 not an error (necessary for create bucket requests)
+        this.req.interceptors.response.use(
+            response => {
+                return response;
+            }, err => {
+                if (err && err.response && err.response.status == 304) {
+                    return err.data;
+                } else {
+                    return Promise.reject(err);
+                }
+            }
+        );
+    }
+
+    info() {
+        return this.req.get('/0/info');
+    }
+
+    createBucket(bucket_id: string, type: string, hostname: string) {
+        return this.req.post('/0/buckets/'+bucket_id, {
+            client: this.clientname,
+            type: type,
+            hostname: hostname,
+        });
+    }
+
+    deleteBucket(bucket_id: string) {
+        return this.req.delete('/0/buckets/'+bucket_id+"?force=1");
+    }
+
+    getBuckets() {
+        return this.req.get("/0/buckets/");
+    }
+
+    getBucketInfo(bucket_id: string) {
+        return this.req.get("/0/buckets/" + bucket_id);
+    }
+
+    // TODO: check what type params should be
+    getEvents(bucket_id: string, params: any) {
+        return this.req.get("/0/buckets/" + bucket_id + "/events", {params: params});
+    }
+
+    // TODO: check what types starttime and endtime are
+    getEventCount(bucket_id: string, starttime: any, endtime: any) {
+        let params = {
+            starttime: starttime,
+            endtime: endtime,
+        }
+        return this.req.get("/0/buckets/" + bucket_id + "/events/count", {params: params});
+    }
+
+    // TODO: Check what type event is
```

Comment:
a javascript object with the format
```
{
  "id": optional int,
  "timestamp": iso8601 string,
  "duration": float (seconds),
  "data": any
}
```

Reply:
The test file suggests that duration is also optional. Is this the case?

Reply:
It is kind of a corner case but it is especially useful in for example heartbeats so I forgot, so yes that is true.

Reply:
If duration is required for normal events (like in `insertEvents`), I guess it would be better to make two interfaces, one for events and one for heartbeats (where it is optional).